crates/sui-types/src/gas.rs

```diff
@@ -389,20 +389,10 @@ impl<'a> SuiGasStatus<'a> {
             .checked_sub(storage_cost)
             .expect("Subtraction overflowed");
         let computation_cost_in_sui = computation_cost.mul(self.computation_gas_unit_price).into();
-        if succeeded {
-            GasCostSummary {
-                computation_cost: computation_cost_in_sui,
-                storage_cost: storage_cost.mul(self.storage_gas_unit_price).into(),
-                storage_rebate: self.storage_rebate.into(),
-            }
-        } else {
-            // If execution failed, no storage creation/deletion will materialize in the store.
-            // Hence they should be 0.
-            GasCostSummary {
-                computation_cost: computation_cost_in_sui,
-                storage_cost: 0,
-                storage_rebate: 0,
-            }
+        GasCostSummary {
+            computation_cost: computation_cost_in_sui,
+            storage_cost: storage_cost.mul(self.storage_gas_unit_price).into(),
+            storage_rebate: self.storage_rebate.into(),
```

Comment:
Clarification question: is it true that, if the transaction fails, the storage cost and storage rebate here are only for the gas objects because we have already reset storage charges for other objects in `reset`?

Reply:
Correct. More precisely, what happens is:
- If the transaction aborts before charging for storage, the storage costs and storage rebate for non-gas objects will be zero
- If the transaction aborts *during* charging for storage, I have added logic to reset the storage cost and storage rebates, and the subsequent code will re-charge storage costs and rebate for the gas object

This is all a bit intricate and could probably be made less fragile via refactoring, but I'm trying to change things as little as possible.